service/src/main/java/com/codeforcommunity/processor/authenticated/ProtectedUserProcessorImpl.java

```diff
@@ -140,6 +143,22 @@ public void disableUserAccount(JWTData userData, int userId) {
     }
     UsersRecord user = db.selectFrom(USERS).where(USERS.ID.eq(userId)).fetchOne();
     if (user == null) {
+      logger.error("`disableUserAccount` tried to disable an account that doesn't exist");
+      throw new UserDoesNotExistException(userId);
+    }
+    user.setDisabled(true);
+    user.store();
+  }
+
+  @Override
+  public void enableUserAccount(JWTData userData, int userId) {
+    if (!userData.isAdmin()) {
+      throw new AdminOnlyRouteException();
+    }
+    UsersRecord user =
+        db.selectFrom(USERS).where(USERS.ID.eq(userId)).and(USERS.DELETED_AT.isNull()).fetchOne();
```

Comment:
We're still missing the check whether the User is already enabled or not. Should be ```db.selectFrom(USERS).where(USERS.ID.eq(userId)).and(USERS.DELETED_AT.isNull()).and(USERS.DISABLED.eq(true)).fetchOne();```